internal/connection/connection_pool.go

```diff
@@ -62,21 +62,34 @@ func (cp *ConnectionPool) Delete(key string) {
 	delete(cp.pool, key)
 }
 
-// Get returns map of connections, filtered with given driverName and optional nodeID.
-func (cp *ConnectionPool) Get(driverName, nodeID string) map[string]*Connection {
-	cp.rwlock.RLock()
-	defer cp.rwlock.RUnlock()
-
+// getByDriverName returns map of connections filtered by driverName. This function
+// must be called with read lock held.
+func (cp *ConnectionPool) getByDriverName(driverName string) map[string]*Connection {
 	newPool := make(map[string]*Connection)
 	for k, v := range cp.pool {
 		if v.DriverName != driverName {
 			continue
 		}
+		newPool[k] = v
+	}
+
+	return newPool
+}
+
+// Get returns map of connections, filtered with given driverName and optional nodeID.
+func (cp *ConnectionPool) Get(driverName, nodeID string) map[string]*Connection {
```

Comment:
Rename this to `GetByNodeID()`? How would a `Get()` with capabilities instead of nodeID be added instead?

Reply:
missed that, renamed it now.